Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/SettingsRepository.kt

```diff
@@ -23,65 +25,12 @@ class SettingsRepository @Inject constructor(
     private val backgroundPrioritization: BackgroundPrioritization
 ) {
 
-    // public mutable live data
-    val isNotificationsEnabled = MutableLiveData(true)
-    val isNotificationsRiskEnabled = MutableLiveData(true)
-    val isNotificationsTestEnabled = MutableLiveData(true)
-    val isManualKeyRetrievalEnabled = MutableLiveData(true)
     val isConnectionEnabled = MutableLiveData(true)
-    val isBluetoothEnabled = MutableLiveData(true)
-    val isLocationEnabled = MutableLiveData(true)
     val isBackgroundJobEnabled = MutableLiveData(true)
-    val isBackgroundPriorityEnabled = MutableLiveData(false)
-    val manualKeyRetrievalTime = MutableLiveData<Long>()
 
-    /**
-     * Get the current notifications state. Only relevant for the ui.
-     *
-     * @see LocalData
-     */
-    fun refreshNotificationsEnabled() {
-        isNotificationsEnabled.value =
-            NotificationManagerCompat.from(context).areNotificationsEnabled()
-    }
-
-    /**
-     * Toggle notifications risk updates.
-     *
-     * @see LocalData
-     */
-    fun toggleNotificationsRiskEnabled() {
-        LocalData.toggleNotificationsRiskEnabled()
-        refreshNotificationsRiskEnabled()
-    }
-
-    /**
-     * Refresh notifications for risk updates with the current shared preferences state.
-     *
-     * @see LocalData
-     */
-    fun refreshNotificationsRiskEnabled() {
-        isNotificationsRiskEnabled.value = LocalData.isNotificationsRiskEnabled()
-    }
-
-    /**
-     * Toggle notifications for test updates in shared preferences and refresh it afterwards.
-     *
-     * @see LocalData
-     */
-    fun toggleNotificationsTestEnabled() {
-        LocalData.toggleNotificationsTestEnabled()
-        refreshNotificationsTestEnabled()
-    }
-
-    /**
-     * Refresh notifications for test updates with the current shared preferences state.
-     *
-     * @see LocalData
-     */
-    fun refreshNotificationsTestEnabled() {
-        isNotificationsTestEnabled.value = LocalData.isNotificationsTestEnabled()
-    }
+    private val internalIsBackgroundPriorityEnabled = MutableStateFlow(false)
+    val isBackgroundPriorityEnabledFlow: Flow<Boolean> = internalIsBackgroundPriorityEnabled
+    val isBackgroundPriorityEnabled = internalIsBackgroundPriorityEnabled.asLiveData()
```

Comment:
I suppose the live data one will be deprecated eventually and you don't rename it to not change a million other occurrences? I'd propose to mark it as deprecated at least with a comment then.

Reply:
Ohh I like the idea, adding  `@Deprecated` :+1: